src/client/app/components/UIOptionsComponent.jsx

```diff
@@ -35,6 +46,24 @@ export default class UIOptionsComponent extends React.Component {
 		this.props.selectMeters(selectedMeters);
 	}
 
+	handleBarDurationChange(value) {
+		this.setState({ barDuration: value });
```

Comment:
It's not obvious that you're putting this in state to delay redux bar changes until the user lets go of the bar. A comment is probably appropriate.